bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudDeviceHandler.java

```diff
@@ -0,0 +1,311 @@
+                }
+                if (val != null) {
+                    // round nearest .5
+                    double v = Math.round(val.doubleValue() * 2) / 2.0;
+                    cmdtoSend.setSetTemperature(v);
+                    cmdtoSend.setEffectiveFlags(EFFECTIVE_FLAG_TEMPERATURE);
+                } else {
+                    logger.debug("Can't convert '{}' to set temperature", command);
+                }
+                break;
+            case CHANNEL_FAN_SPEED:
+                cmdtoSend.setSetFanSpeed(((DecimalType) command).intValue());
+                cmdtoSend.setEffectiveFlags(EFFECTIVE_FLAG_FAN_SPEED);
+                break;
+            case CHANNEL_VANE_VERTICAL:
+                cmdtoSend.setVaneVertical(((DecimalType) command).intValue());
+                cmdtoSend.setEffectiveFlags(EFFECTIVE_FLAG_VANE_VERTICAL);
+                break;
+            case CHANNEL_VANE_HORIZONTAL:
+                cmdtoSend.setVaneHorizontal(((DecimalType) command).intValue());
+                cmdtoSend.setEffectiveFlags(EFFECTIVE_FLAG_VANE_HORIZONTAL);
+                break;
+            default:
+                logger.debug("Read only or unknown channel {}, skip update", channelUID);
+        }
+
+        if (cmdtoSend.getEffectiveFlags() > 0) {
+            cmdtoSend.setHasPendingCommand(true);
+            cmdtoSend.setDeviceID(config.deviceID);
+            try {
+                DeviceStatus newDeviceStatus = melCloudHandler.sendDeviceStatus(cmdtoSend);
+                updateChannels(newDeviceStatus);
+            } catch (MelCloudCommException e) {
+                logger.warn("Command '{}' to channel '{}' failed, reason {}. ", command, channelUID, e.getMessage(), e);
+            }
+        } else {
+            logger.debug("Nothing to send");
+        }
+    }
+
+    private DeviceStatus getDeviceStatusCopy(DeviceStatus deviceStatus) {
+        DeviceStatus copy = new DeviceStatus();
+        synchronized (this) {
+            copy.setPower(deviceStatus.getPower());
+            copy.setOperationMode(deviceStatus.getOperationMode());
+            copy.setSetTemperature(deviceStatus.getSetTemperature());
+            copy.setSetFanSpeed(deviceStatus.getSetFanSpeed());
+            copy.setVaneVertical(deviceStatus.getVaneVertical());
+            copy.setVaneHorizontal(deviceStatus.getVaneHorizontal());
+            copy.setEffectiveFlags(deviceStatus.getEffectiveFlags());
+            copy.setHasPendingCommand(deviceStatus.getHasPendingCommand());
+            copy.setDeviceID(deviceStatus.getDeviceID());
+        }
+        return copy;
+    }
+
+    private void startAutomaticRefresh() {
+        if (refreshTask == null || refreshTask.isCancelled()) {
+            Runnable runnable = () -> getDeviceDataAndUpdateChannels();
+            refreshTask = scheduler.scheduleWithFixedDelay(runnable, 1, config.pollingInterval, TimeUnit.SECONDS);
```

Comment:
No need for the runnable variable:
```suggestion
            refreshTask = scheduler.scheduleWithFixedDelay(this:getDeviceDataAndUpdateChannels, 1, config.pollingInterval, TimeUnit.SECONDS);
```